kotlin/reflect/src/main/java/com/squareup/moshi/kotlin/reflect/KotlinJsonAdapter.kt

```diff
@@ -164,7 +174,12 @@ internal class KotlinJsonAdapter<T>(
   }
 }
 
-class KotlinJsonAdapterFactory : JsonAdapter.Factory {
+enum class FailureStrategy {
+  FAIL_FAST, ACCUMULATE_ERRORS
```

Comment:
While the names are fairly clear, I would still document the behavior on these values.